src/CMakeLists.txt

```diff
@@ -31,40 +43,79 @@ ExternalProject_Add(gflags-proj
+set(GFLAGS_LIBRARY_PATH ${GFLAGS_ROOT}/lib/libgflags${GFLAGS_WIN_LIB_SUFFIX}.a)
+set(GFLAGS_DEBUG_LIBRARY_PATH ${GFLAGS_ROOT}/lib/libgflags_debug.a)
 set(GFLAGS_LIBRARY gflags)
+
 add_library(${GFLAGS_LIBRARY} STATIC IMPORTED)
+
+generate_export_header(${GFLAGS_LIBRARY})
+
 set_target_properties(${GFLAGS_LIBRARY} PROPERTIES
         IMPORTED_LOCATION ${GFLAGS_LIBRARY_PATH}
         IMPORTED_LOCATION_DEBUG ${GFLAGS_DEBUG_LIBRARY_PATH}
-        INTERFACE_LINK_LIBRARIES ${CMAKE_THREAD_LIBS_INIT})
+        INTERFACE_LINK_LIBRARIES Threads::Threads)
+
 add_dependencies(${GFLAGS_LIBRARY} gflags-proj)
 
-# Setup mgclient library
-find_library(MGCLIENT_LIBRARY mgclient)
-find_path(MGCLIENT_INCLUDE_DIRS mgclient.h)
+ExternalProject_Add(mgclient-proj
+        PREFIX            mgclient
+        GIT_REPOSITORY    https://github.com/memgraph/mgclient.git
+        GIT_TAG           master
+        CMAKE_ARGS        "-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>"
+                          "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
+        INSTALL_DIR       "${PROJECT_BINARY_DIR}/mgclient")
 
-if (MGCLIENT_LIBRARY AND MGCLIENT_INCLUDE_DIRS)
-  message(STATUS "Found mgclient: " ${MGCLIENT_LIBRARY} " "
-          ${MGCLIENT_INCLUDE_DIRS})
-else()
-  message(FATAL_ERROR "Could not find mgclient")
-endif()
+ExternalProject_Get_Property(mgclient-proj install_dir)
+set(MGCLIENT_ROOT ${install_dir})
+set(MGCLIENT_INCLUDE_DIRS ${MGCLIENT_ROOT}/include
+        CACHE INTERNAL "Path to mgclient include directory")
+set(MGCLIENT_LIBRARY_PATH ${MGCLIENT_ROOT}/lib/libmgclient.a)
+set(MGCLIENT_LIBRARY mgclient)
+
+add_library(${MGCLIENT_LIBRARY} STATIC IMPORTED GLOBAL)
+
+generate_export_header(${MGCLIENT_LIBRARY})
 
-add_compile_options(-Wall -Wextra -pedantic -Werror)
+set_target_properties(${MGCLIENT_LIBRARY} PROPERTIES
+  IMPORTED_LOCATION ${MGCLIENT_LIBRARY_PATH}
+  INTERFACE_LINK_LIBRARIES Threads::Threads)
+
+add_dependencies(${MGCLIENT_LIBRARY} mgclient-proj)
+
+include_directories(${MGCLIENT_INCLUDE_DIRS})
+
+add_subdirectory(utils)
+add_dependencies(utils replxx mgclient)
+
+# replxx performs narrowing conversion on Windows
+if (${CMAKE_SYSTEM_NAME} MATCHES "Windows") 
+  set(NO_NARROWING_CONVERSION_ERROR -Wno-narrowing)
+endif()
+add_compile_options(-Wall -Wextra -pedantic -Werror ${NO_NARROWING_CONVERSION_ERROR})
 
 add_executable(mgconsole main.cpp)
-target_include_directories(mgconsole PRIVATE
+target_compile_definitions(mgconsole
+        PRIVATE
+        UTILS_STATIC_DEFINE
+        MGCLIENT_STATIC_DEFINE
+        GFLAGS_STATIC_DEFINE)
+target_include_directories(mgconsole
+        PRIVATE
         ${GFLAGS_INCLUDE_DIRS}
         ${MGCLIENT_INCLUDE_DIRS}
+        ${REPLXX_INCLUDE_DIRS}
         ${CMAKE_CURRENT_BINARY_DIR})
-target_link_libraries(mgconsole ${MGCLIENT_LIBRARY} ${GFLAGS_LIBRARY}
-        ${READLINE_LIBRARY} stdc++fs)
+target_link_libraries(mgconsole
+        PRIVATE
+        ${GFLAGS_LIBRARY}
+        replxx
```

Comment:
does mgconsole need replxx?
If it does it should go after utils.

Reply:
It doesn't. It's removed